{{cookiecutter.project_name}}/monitoring/model_monitoring_setup_template.py

```diff
@@ -0,0 +1,278 @@
+# MAGIC 
+# MAGIC #### Example derived per-column summary statistic
+# MAGIC ```
+# MAGIC dm.Metric(
+# MAGIC    metric_type="derived”,
+# MAGIC    metric_name="root_mean_square",
+# MAGIC    input_columns=["price", "invoice"],
+# MAGIC    metric_definition="sqrt(squared_avg)",
+# MAGIC    output_type=T.DoubleType()
+# MAGIC )
+# MAGIC ```
+# MAGIC This example computes the square root of the squared_avg metric defined above for the price and invoice columns. Notice that since this is a derived metric, it does not reference the primary table data and instead is defined in terms of the squared_avg aggregate metric. The output will be available as a new column in the analysis metrics table.
+# MAGIC 
+# MAGIC #### Example model quality score
+# MAGIC ```
+# MAGIC dm.Metric(
+# MAGIC    metric_type="aggregate”,
+# MAGIC    metric_name="weighted_error",
+# MAGIC    input_columns=[":table"],
+# MAGIC    metric_definition="""avg(CASE 
+# MAGIC      WHEN {{prediction_col}} = {{label_col}} THEN 0
+# MAGIC      WHEN {{prediction_col}} != {{label_col}} AND critical=TRUE THEN 2
+# MAGIC      ELSE 1 END))""",
+# MAGIC    output_type=T.DoubleType()
+# MAGIC )
+# MAGIC ```
+# MAGIC This example computes a weighted model quality score that penalizes when the prediction does not equal the ground truth label when the critical column is true. Because it’s defined on the raw columns (prediction and label), it’s defined as an aggregate metric. The ":table" column indicates that this metric applies to multiple columns, rather than being specific to a single input-column. The {{prediction_col}} and {{label_col}} Jinja parameters will be replaced with the name of the prediction and ground truth label columns for the monitor.
+# MAGIC 
+# MAGIC #### Example model drift metric
+# MAGIC ```
+# MAGIC dm.Metric(
+# MAGIC    metric_type="drift”,
+# MAGIC    metric_name="error_rate_delta",
+# MAGIC    input_columns=[":table"],
+# MAGIC    metric_definition="{{cur_df}}.weighted_error - {{base_df}}.weighted_error",
+# MAGIC    output_type=T.DoubleType()
+# MAGIC )
+# MAGIC ```
+# MAGIC This example computes a drift metric corresponding to the change in the weighted error score. The {{cur_df}} and {{base_df}} parameters allow the metric to reference the weighted_error metrics from both the current window and as baseline comparison window: either the baseline data or the previous time window. Unlike the other metrics, drift metrics are stored in the drift metrics table rather than the analysis metrics table.
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC 
+# MAGIC ## Alerts
+# MAGIC 
+# MAGIC Data Monitoring computes important quality - and performance - indicators (data/model quality, drift, etc.), and a typical next step is to set up alerts for when these indicators look "abnormal". Because the performance metrics are stored in Delta tables, we recommend using SQL alerts (AWS | Azure | GCP) as the mechanism to set this up.
```

Comment:
I'm guessing you lifted a lot of this from our user guide - do you mind taking another pass to clean it up a bit? For example, the `(AWS | Azure | GCP)` are doc links from the user guide. Those are not relevant here, or if they are the links are not there (and we are not allowed to link to other clouds from Azure workspaces).

There are also dead hyperlinks I see in other places.

Reply:
Actually, another question I have is do we need to in-line all of this information from the user guide, which may be out of date? Can we instead just provide the user guide to the customer, which we will keep live and up to date (with links)